File: matrices_multidimensionales.py
def menu():
    personas = []
    
    while True:
        print("\n--- Menú ---")
        print("1. Ingresar nueva persona")
        print("2. Mostrar todos los datos")
        print("3. Filtrar por DNI")
        print("4. Salir")
        
        opcion = input("Elige una opción: ")
        
        if opcion == "1":
            nombre = input("Ingresa el nombre: ")
            apellido = input("Ingresa el apellido: ")
            dni = input("Ingresa el DNI: ")
            telefonos = [t.strip() for t in input("Ingresa los teléfonos separados por comas: ").split(",")]
            hijos = [h.strip() for h in input("Ingresa los nombres de los hijos separados por comas: ").split(",") if h.strip() != ""]
            
            personas.append([nombre, apellido, dni, telefonos, hijos])
            print("Persona agregada con éxito!")
        
        elif opcion == "2":
            print("\nDatos ingresados:")
            print(personas)
            for p in personas:
                print(f"{p[0]} {p[1]} - DNI: {p[2]}, Teléfonos: {len(p[3])}, Hijos: {len(p[4])}")
        
        elif opcion == "3":
            dni_filtrar = input("Ingresa el DNI para filtrar: ")
            encontrada = None
            
            for p in personas:
                if p[2] == dni_filtrar:
                    encontrada = p
                    break
            
            if encontrada:
                print(f"\nDatos de {encontrada[0]} {encontrada[1]}:")
                print(f"DNI: {encontrada[2]}")
                print(f"Teléfonos: {len(encontrada[3])} teléfono(s)")
                print(f"Hijos: {len(encontrada[4])}")
            else:
                print("No se encontró una persona con ese DNI.")
        
        elif opcion == "4":
            print("Saliendo...")
            break
        
        else:
            print("Opción no válida, intenta de nuevo.")

if __name__ == "__main__":
    menu()

File: matrices_multidimensionales.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const personas = [];

function ask(question) {
  return new Promise(resolve => rl.question(question, resolve));
}

async function menu() {
  while (true) {
    console.log("\n--- Menú ---");
    console.log("1. Ingresar nueva persona");
    console.log("2. Mostrar todos los datos");
    console.log("3. Filtrar por DNI");
    console.log("4. Salir");
    
    const opcion = await ask("Elige una opción: ");
    
    if (opcion === "1") {
      const nombre = await ask("Ingresa el nombre: ");
      const apellido = await ask("Ingresa el apellido: ");
      const dni = await ask("Ingresa el DNI: ");
      const telefonos = (await ask("Ingresa los teléfonos separados por comas: ")).split(",").map(t => t.trim());
      const hijos = (await ask("Ingresa los nombres de los hijos separados por comas: ")).split(",").map(t => t.trim()).filter(t => t !== "");
      
      personas.push([nombre, apellido, dni, telefonos, hijos]);
      console.log("Persona agregada con éxito!");
      
    } else if (opcion === "2") {
      console.log("\nDatos ingresados:");
      console.log(personas);
      personas.forEach(p => {
        console.log(`${p[0]} ${p[1]} - DNI: ${p[2]}, Teléfonos: ${p[3].length}, Hijos: ${p[4].length}`);
      });
      
    } else if (opcion === "3") {
      const filtrarDNI = await ask("Ingresa el DNI para filtrar: ");
      const persona = personas.find(p => p[2] === filtrarDNI);
      
      if (persona) {
        console.log(`\n${persona[0]} ${persona[1]} - DNI: ${persona[2]}, Teléfonos: ${persona[3].length}, Hijos: ${persona[4].length}`);
      } else {
        console.log("No se encontró una persona con ese DNI.");
      }
      
    } else if (opcion === "4") {
      console.log("Saliendo...");
      rl.close();
      break;
      
    } else {
      console.log("Opción no válida, intenta de nuevo.");
    }
  }
}

menu();
